modules/dnn/src/layers/eltwise_layer.cpp

```diff
@@ -210,6 +210,45 @@ class EltwiseLayerImpl CV_FINAL : public EltwiseLayer
         return false;
     }
 
+    virtual bool tryFuse(std::vector< Ptr<dnn::Layer> >& bottoms, Ptr<dnn::Layer>& top) CV_OVERRIDE
+    {
+        // Either all input layers are fusing or nothing.
+        for (int i = 1; i < bottoms.size(); i++)
+        {
+            if (bottoms[0]->type != bottoms[i]->type)
+                return false;
+        }
+
+        std::vector< Ptr<dnn::Layer> > fakeInputs;
+        bool isFused = bottoms[0]->tryFuse(fakeInputs, top);
+        if (isFused)
+        {
+            Mat w, b;
+            top->getScaleShift(w, b);
+            if (!w.empty())
+            {
+                LayerParams weights_lp;
+                weights_lp.blobs.push_back(w);
+                Ptr<Layer> weightsLayer = ScaleLayer::create(weights_lp);
+                weightsLayer->finalize(std::vector<Mat>(1), std::vector<Mat>(1));
+                //  layer 1        layer 2        layer 3
+                //      \             |             X
+                //        \           |           X
+                //          \         |         X
+                //            \       |       X
+                //              \     |     X
+                //                \   |   X
+                //                fuse layer
+                //  We need to check that every layer will be fused
+                for (size_t i = 1; i < bottoms.size(); i++)
+                {
+                    CV_Assert(bottoms[i]->tryFuse(bottoms, weightsLayer));
```

Comment:
Do we really need `CV_Assert()` here? This is "try" function, not validation.

In case of assertion (non-supported case) network is left in non-consistent (broken) state.
Some transaction-based fusing should be implemented instead.